### PR TITLE
tools: openSUSE fixes

### PR DESCRIPTION
**Pull request by lnussel**
- Require group wheel
- Add permission macros
- Create tmpfiles in %post install
- Don't require systemd. This allows to build smaller containers

### CODE REVIEW

**Comment by lnussel** [comment]
in openSUSE the systemd macros are set up to not fail if systemd isn't present. Not sure that is the case in Fedora too so guarded by %if

File: tools/cockpit.spec [hunk 1]
```diff
@@ -454,11 +454,16 @@ Requires: glib2 >= 2.50.0
 Conflicts: firewalld < 0.6.0-1
 Recommends: sscg >= 2.3
 Recommends: system-logos
-Requires: systemd >= 235
 Suggests: sssd-dbus
+%if !0%{?suse_version}
```
Comment (martinpitt) [inline tools/cockpit.spec:458]: Do you mind flipping this condition, so that we keep all the suse special cases in the same style?

File: tools/cockpit.spec [hunk 2]
```diff
@@ -454,11 +454,16 @@ Requires: glib2 >= 2.50.0
...
+Requires: systemd >= 235
```
Comment (martinpitt) [inline tools/cockpit.spec:459]: We need that in Fedora/Debian/etc. to get proper values for 

    systemdunitdir=$($PKG_CONFIG systemd --variable=systemdsystemunitdir)

how does that work in OpenSUSE? is systemd.pc shipped in a different package? I'd also be okay with generally dropping that build dependency and hardcoding `--with-systemdunitdir=/usr/lib/systemd/`, but it feels wrong to do that. What's the problem with this extra build dep?

File: tools/cockpit.spec [hunk 3]
```diff
@@ -454,11 +454,16 @@ Requires: glib2 >= 2.50.0
...
 Requires(post): systemd
 Requires(preun): systemd
 Requires(postun): systemd
+%else
+Requires: group(wheel)
+Requires(pre): permissions
```
Comment (martinpitt) [inline tools/cockpit.spec:465]: What does that do? Can you please give a rationale to the commit message?

Cockpit doesn't really care about the `wheel` group. I just see that we still reference it here, though:

    %attr(775, -, wheel) %{_localstatedir}/lib/cockpit

We shouldn't create that directory at all any more, I'll send a PR to eliminate it. If  it was just about that, then please drop this bit.

File: tools/cockpit.spec [hunk 4]
```diff
@@ -516,6 +521,10 @@ getent group cockpit-wsinstance >/dev/null || groupadd -r cockpit-wsinstance
 getent passwd cockpit-wsinstance >/dev/null || useradd -r -g cockpit-wsinstance -d /nonexisting -s /sbin/nologin -c "User for cockpit-ws instances" cockpit-wsinstance
 
 %post ws
+%if 0%{?suse_version}
+%set_permissions %{_libexecdir}/cockpit-session
```
Comment (martinpitt) [inline tools/cockpit.spec:525]: What does that do in addition to

    %attr(4750, root, cockpit-wsinstance) %{_libexecdir}/cockpit-session

?

File: tools/cockpit.spec [hunk 5]
```diff
@@ -516,6 +521,10 @@ getent group cockpit-wsinstance >/dev/null || groupadd -r cockpit-wsinstance
...
+%tmpfiles_create cockpit-tempfiles.conf
```
Comment (martinpitt) [inline tools/cockpit.spec:526]: This would  be an excellent thing to do on Fedora/RHEL as well. Do you mind moving it out of the `%if`? That would also not hurt to be a separate commit then.

File: tools/cockpit.spec [hunk 6]
```diff
@@ -527,6 +536,11 @@ test -f %{_bindir}/firewall-cmd && firewall-cmd --reload --quiet || true
 %systemd_postun_with_restart cockpit.socket
 %systemd_postun_with_restart cockpit.service
 
+%if 0%{?suse_version}
+%verifyscript ws
+%verify_permissions -e %{_libexecdir}/cockpit-session
```
Comment (martinpitt) [inline tools/cockpit.spec:541]: I'm also curious what this thing does?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thank you!

File: tools/cockpit.spec (hunk 3 above)
Comment (martinpitt) [reply tools/cockpit.spec:465]: I sent PR #14588 to clean up that bit.

File: tools/cockpit.spec (hunk 1 above)
Comment (lnussel) [reply tools/cockpit.spec:458]: sure. will do

File: tools/cockpit.spec (hunk 2 above)
Comment (lnussel) [reply tools/cockpit.spec:459]: This one is about the run time requirement. The BuildRequires stays of course :)

File: tools/cockpit.spec (hunk 3 above)
Comment (lnussel) [reply tools/cockpit.spec:465]: I guess it's about that so will remove

File: tools/cockpit.spec (hunk 4 above)
Comment (lnussel) [reply tools/cockpit.spec:525]: it's the SUSE runtime permission handling. we have permission profiles so can change setuid bits according to run time config. https://github.com/openSUSE/permissions/

File: tools/cockpit.spec (hunk 5 above)
Comment (lnussel) [reply tools/cockpit.spec:526]: ok

File: tools/cockpit.spec (hunk 6 above)
Comment (lnussel) [reply tools/cockpit.spec:541]: Since rpm can't verify the permissions (which makes me realize the patch was incomplete) there has to be a verifyscript that checks if file permissions match the config.

File: tools/cockpit.spec (hunk 2 above)
Comment (martinpitt) [reply tools/cockpit.spec:459]: Oh, right. systemd is a hard requirement for cockpit.socket/cockpit.service, but I see the container build use case (we actually do that upstream as well, https://hub.docker.com/r/cockpit/ws/).

The main point of this was to make sure that it's not running against a too old systemd, but all our supported OSes now have a sufficiently new one. So it's fine for me to simply drop this Requires: completely.

Comment (lnussel) [reply tools/cockpit.spec:459]: Ok. In such a case a conflicts with < someversion would also work.

**Comment by lnussel** [comment]
#14599 and #14600

**Comment by martinpitt** [comment]
#14599 and #14600 landed now, so this needs a rebase and some cleanup.

**Comment by lnussel** [comment]
Will do after my vacations, thanks!
